azurelinuxagent/ga/exthandlers.py

```diff
@@ -871,6 +923,35 @@ def collect_ext_status(self, ext):
 
         return ext_status
     
+    def get_ext_handling_status(self, ext):
+        seq_no, ext_status_file = self.get_status_file_path(ext)
+        if seq_no < 0 or ext_status_file is None:
+            return None
+
+        # Missing status file is considered a non-terminal state here
+        # so that extension sequencing can wait until it becomes existing
+        if not os.path.exists(ext_status_file):
+            status = "warning"
```

Comment:
we should change this status... this gives the impression that the extension is reporting a warning when in actuality there is no status file. I would suggest return something like "status file not found" or similar

Reply:
This value is verified against a set of valid values. I think we should adhere to those values. Also it is clearly commented above that condition in which it can occur.

Reply:
My concern is that if reach this path, then the status in this message:

"Extension {0} did not reach a terminal state within the allowed timeout. Last status was {1}".

will be "warning". When looking at the log, it'll seemed like the extension actually reported a warning, when in actuality  it didn't report any status. That may be confusing.

Reply:
We need to pass up the entire status, not just the status field, to give a more complete log message and add a `__str__` to `ExtStatus` so that it logs something sensible